Update index.md to add a tool for Django

StuartMacKay opened the pull request:
Add an entry for Crispy Forms GDS, a template pack for Django Crispy Forms to generate forms and pages for the GOV.UK Frontend for Django based web sites.

owenatgov reviewed (commented):
Hi @StuartMacKay. Sorry this has taken so long to respond to. Thanks for the contribution!

I'm happy with this in proncipal, however there are a couple of things I'd like to address regarding the the tool to ensure it meets requirements for [our resource and tools contribution criteria](https://design-system.service.gov.uk/community/contribution-criteria/#developing-a-community-resource-or-tool):

1. In the readme it lists 2.x (the latest version from [tags](https://github.com/wildfish/crispy-forms-gds/tags) being 2.0.1) as supporting govuk-frontend v5 to v5.8. I'm unclear on what this means. Is this clarified somewhere?
2. The readme is out of date and suggets that the tool is still in development. Could this be updated please?
3. Would you also be willing to add the quote at the bottom of the linked contribution criteria OR clearly indicate somewhere that this is a community resource and not an official part of the GOV.UK Design System?

I'm happy to raise these and 1 or more issues on the repo if that's easier. Once these are resolved I reckon this can get added no problem.

StuartMacKay commented:
@owenatgov I agree the version numbering and wording is confusing. 

As a slight aside, the list of apparently active versions was written this way, as at least one group in the Dept. of Business and Trade are heavily invested in using this library. I don't have any details, so the compatibility matrix was written with them in mind so they can see where they are now, what versions of Django are supported, and what they have to do to upgrade. The reason Django comes into the picture is that the way HTML attributes are generated by forms changed slightly with the release of Django 5.0, breaking the crispy-forms-gds tests. Rather than add lots of conditionals, complicating the code, it was easier to simply release 2.0. This also offered the opportunity to end support for older versions of the Design System, Django Crispy Forms and Django itself. Crispy-forms-gds is platform agnostic (it simply generates HTML, so it likely compatible with more or less everything), but life is easier is there is a well defined set of targets to develop and test against.

So to address the confusion, I changed the following:

1. I removed the entry for 0.x. It is probably only relevant to old projects at Business and Trade and if they see their version is missing then they'll know it's time to upgrade.
2. I added a "maintenance" label to 1.x and "current" to 2.x. That should make clear the relationship and which version to use for new projects.
3. I removed the reference to development in the text following the table. The text now states that not all components described by the Government Design System are currently implemented, so they will be added in future releases. Version 1.x will be updated until April 2026 when Django 4.2 LTS changes to "end of life". Only bugs will be fixed after that. New releases of 2.x will be made as components are added.

For the note about this being a community project. The text at the bottom of the README states exactly that:

"Crispy Forms GDS is a community tool of the GOV.UK Design System. The Design System team is not responsible for it and cannot support you with using it. ...."

However I changed community to "community-supported". 

I also added text to say that the easiest way to get in touch with Wildfish is to open an issue on our GitHub repository.

That should address all the point you raised.

owenatgov reviewed (approved):
@StuartMacKay This took ages again 🙈 Very sorry. I'm happy with this! Just needs a 2nd review and it can get merged.

romaricpascal commented:
Just did a force push to the branch in the hopes to trigger the required checks so we can merge it 🤞🏻
